src/auth/services/users.service.ts

```diff
@@ -144,4 +162,112 @@ export class UserService {
       }),
     );
   }
+
+  async findOthersProfiles(profileFilter: GetProfilesFilterDto) {
+    const { search } = profileFilter;
+
+    const query = this.userRepository.createQueryBuilder('user');
+
+    if (search) {
+      query.where(
+        '(LOWER(user.firstname) LIKE LOWER(:search) OR LOWER(user.lastname) LIKE LOWER(:search))',
+        {
+          search: `%${search}%`,
+        },
+      );
+    }
+
+    const profiles = await query.getMany();
+    return profiles;
+  }
+
+  async updateProfile(
+    user: User,
+    updateProfileDto: UpdateProfileDto,
+  ): Promise<User> {
+    const profile = await this.userRepository.findOne({
```

Comment:
Tại sao findOne ở đây lại cần dùng đến where e?

Reply:
Dạ e làm hơi cầu kỳ. Chỉ cần `const profile = await this.userRepository.findOne(user.id);` là ok r ạ